### PR TITLE
MULE-16987: Interception API: Parameters of smart connector not intercepted

### PR DESCRIPTION
**Pull request by emariotti3**
…cepted.

### CODE REVIEW

**Comment by emariotti3** [comment]
--validate

File: integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java [hunk 1]
```diff
@@ -476,17 +464,9 @@ public void scUsingScOperation() throws Exception {
...
 
-    assertThat(proxyModuleOperationChain.getParameters().keySet(), containsInAnyOrder("moduleName", "moduleOperation"));
-    assertThat(proxyModuleOperationChain.getParameters().get("moduleName").resolveValue(), is("module-using-sc"));
-    assertThat(proxyModuleOperationChain.getParameters().get("moduleOperation").resolveValue(),
-               is("proxy-set-payload-hardcoded"));
-
-    assertThat(innerModuleOperationChain.getParameters().keySet(), containsInAnyOrder("moduleName", "moduleOperation"));
-    assertThat(innerModuleOperationChain.getParameters().get("moduleName").resolveValue(), is("module-using-core"));
-    assertThat(innerModuleOperationChain.getParameters().get("moduleOperation").resolveValue(), is("set-payload-hardcoded"));
+    assertThat(proxyModuleOperationChain.getParameters().keySet(), containsInAnyOrder("targetValue"));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java]: keep an assertion on `innerModuleOperationChain `

File: integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java [hunk 2]
```diff
@@ -450,16 +443,11 @@ public void scEchoOperationFlowRef() throws Exception {
     assertThat(interceptionParameters, hasSize(3));
 
     InterceptionParameters flowRef = interceptionParameters.get(0);
-    InterceptionParameters moduleOperationChain = interceptionParameters.get(1);
     InterceptionParameters setPayloadOperation = interceptionParameters.get(2);
 
     assertThat(flowRef.getParameters().keySet(), containsInAnyOrder("name", "targetValue"));
     assertThat(flowRef.getParameters().get("name").resolveValue(), is("scEchoOperation"));
 
-    assertThat(moduleOperationChain.getParameters().keySet(), containsInAnyOrder("moduleName", "moduleOperation"));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java]: keep an assertion on `moduleOperationChain`

File: integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java [hunk 3]
```diff
@@ -426,13 +424,8 @@ public void scEchoOperation() throws Exception {
     List<InterceptionParameters> interceptionParameters = HasInjectedAttributesInterceptor.interceptionParameters;
     assertThat(interceptionParameters, hasSize(2));
 
-    InterceptionParameters moduleOperationChain = interceptionParameters.get(0);
     InterceptionParameters setPayloadOperation = interceptionParameters.get(1);
 
-    assertThat(moduleOperationChain.getParameters().keySet(), containsInAnyOrder("moduleName", "moduleOperation"));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java]: keep an assertion on `moduleOperationChain`

**Pull request merged**